src/Lykke.Service.Decred.Api.Services/TransactionBroadcastService.cs

```diff
@@ -0,0 +1,191 @@
+        private readonly ITransactionRepository _txRepo;
+
+        private readonly INosqlRepo<BroadcastedTransaction> _broadcastTxRepo;
+        private readonly INosqlRepo<BroadcastedTransactionByHash> _broadcastTxHashRepo;
+        private readonly INosqlRepo<BroadcastedOutpoint> _broadcastedOutpointRepo;
+
+        public TransactionBroadcastService(
+            IDcrdClient dcrdClient,
+            IBlockRepository blockRepository,
+            ITransactionRepository txRepo,
+            INosqlRepo<BroadcastedOutpoint> broadcastedOutpointRepo,
+            INosqlRepo<BroadcastedTransaction> broadcastTxRepo,
+            INosqlRepo<BroadcastedTransactionByHash> broadcastTxHashRepo)
+        {
+            _dcrdClient = dcrdClient;
+            _blockRepository = blockRepository;
+            _txRepo = txRepo;
+
+            _broadcastTxRepo = broadcastTxRepo;
+            _broadcastTxHashRepo = broadcastTxHashRepo;
+            _broadcastedOutpointRepo = broadcastedOutpointRepo;
+        }
+
+        private string[] GetOutpointKeysForRawTransaction(string hexTransaction)
+        {
+            var txBytes = HexUtil.ToByteArray(hexTransaction);
+            return GetOutpointKeysForRawTransaction(txBytes);
+        }
+
+        private string[] GetOutpointKeysForRawTransaction(byte[] txBytes)
+        {
+            var tx = Transaction.Deserialize(txBytes);
+            return tx.Inputs.Select(input => input.PreviousOutpoint.ToString()).ToArray();
+        }
+
+        /// <summary>
+        /// Broadcasts a signed transaction to the Decred network
+        /// </summary>
+        /// <param name="operationId"></param>
+        /// <param name="hexTransaction"></param>
+        /// <returns></returns>
+        /// <exception cref="TransactionBroadcastException"></exception>
+        public async Task Broadcast(Guid operationId, string hexTransaction)
+        {
+            if (operationId == Guid.Empty)
+                throw new BusinessException(ErrorReason.BadRequest, "Operation id is invalid");
+            if (string.IsNullOrWhiteSpace(hexTransaction))
+                throw new BusinessException(ErrorReason.BadRequest, "SignedTransaction is invalid");
+
+            var txBytes = HexUtil.ToByteArray(hexTransaction);
+            var msgTx = new MsgTx();
+            msgTx.Decode(txBytes);
+
+            // If the operation exists in the cache, throw exception
+            var cachedResult = await _broadcastTxRepo.GetAsync(operationId.ToString());
+            if (cachedResult != null)
+                throw new BusinessException(ErrorReason.DuplicateRecord, "Operation already broadcast");
+
+            // Submit the transaction to the network via dcrd
+            var result = await _dcrdClient.SendRawTransactionAsync(hexTransaction);
+            if (result.Error != null)
+                throw new TransactionBroadcastException($"[{result.Error.Code}] {result.Error.Message}");
+
```

Comment:
What if service fail here (or in foreach loop, or before SaveBroadcastedTransaction called)? It seems, that transaction will be broadcasted, but will never be saved to repo. Also, I think, that it will be impossible to recall this method for the same transaction. If I'm right, please, make this method idempotent.